vignettes/slopes.Rmd

```diff
@@ -15,103 +14,390 @@ knitr::opts_chunk$set(
+- `z_start()` retrieves the elevation value of the first linestring vertice
+- `z_end()` retrieves the elevation value of the last linestring vertice
+- `z_mean()` retrieves the elevation mean value 
+- `z_max()`  retrieves the elevation max value 
+- `z_min()`  retrieves the elevation min value 
+
+### Distance
+
+- `sequential_dist()` Calculate the sequential distances between sequential coordinate pairs
+
+### Slope
 
-$$
-s = \Delta z/l
-$$
+- `slope_vector()` calculates the slopes associated with consecutive elements in one dimensional distance and associated elevations.
+- `slope_distance()` calculates the slopes associated with consecutive distances and elevations.
+- `slope_distance_mean()` calculates the mean average slopes associated with consecutive distances and elevations.
+- `slope_distance_weighted()` calculates the slopes associated with consecutive distances and elevations, with the mean value associated with each set of distance/elevation vectors weighted in proportion to the distance between each elevation measurement, so longer sections have proportionally more influence on the resulting gradient estimate.
 
-Depending on the purpose of application, it might me relevant to understand how hilliness is estimated.  
+- `slope_raster()` Calculate slopes of linestrings based on local raster map
+- `slope_matrix()` Calculate the gradient of line segments from a 3D matrix of coordinates
+- `slope_matrix_weighted()` Calculate the weighted gradient of line segments from a 3D matrix of coordinates
 
-# Implementation in the slopes package
+- `slope_xyz()` Calculates the slope associated with linestrings that have xyz coordinates
 
-Using the `slopes` package we can estimate the gradient of individual road segments. When these segments are combined into routes, we then need a means of assessing the hilliness of the entire route. A range of indices can be used to represent route hilliness. The choice of which index is most appropriate may be context dependent.
+### Plot
 
-To help research slopes using open source software for reproducibility, we have developed the slopes package.
-After you have installed it, load it as follows:
+- `plot_dz()` Plot a digital elevation profile based on xyz data
+- `plot_slope()` Plots the slope profile associated with a linestring with base R graphics
 
-```{r setup}
+## Package datasets
+
+The `slopes` package comes with some datasets to play with:
+
+**Linestrings:**
+
+- `lisbon_road_segment`: a single road segment in Lisbon (XY)
+- `lisbon_route`: a route with some variation in elevation in Lisbon (XY)
+- `cyclestreets_route`: a bike route in Leeds (XY)
+
+**Road network:**
+
+- `lisbon_road_network`: a sample of road segments in downtown Lisbon
+- `magnolia_xy`: a sample of road segments in center Seattle, in the Magnolia neighborhood
+
+**Digital elevation model (DEM):**
+
+- `dem_lisbon_raster` a DEM of downtown Lisbon (EPSG:3763)
```

Comment:
Suggestion: rename it to `lisbon_dem_raster` so it's consistent. Can do post merge.